File: programmers/연습문제/level2/기능개발.py
from math import ceil

def solution(progresses, speeds):
    answer = []
    progresses = [ceil((100 - a)/b) for a, b in zip(progresses, speeds)]
    pos = 0
    
    for i in range(len(progresses)):
        if progresses[i] > progresses[pos]:
            answer.append(i - pos)
            pos = i
            
    answer.append(len(progresses) - pos)
    return answer

File: programmers/연습문제/level2/기능개발.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>
#include <stack>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int totalWork = progresses.size();
    
    vector<int> days;
    for (int i = 0; i < totalWork; i++) {
        days.push_back(ceil((float) (100 - progresses[i])/speeds[i]));
    }
    
    stack<int> st;
    int first;
    for (int d : days) {
        if (st.empty()) {
            first = d;
            st.push(d);
        } else {
            if (first >= d) {
                st.push(d);
            } else {
                answer.push_back(st.size());
                while (!st.empty()) {
                    st.pop();
                }
                first = d;
                st.push(d);
            }
        }
    }
    answer.push_back(st.size());
    return answer;
}